File: Arrays/Subarray_with_Given_Sum.cpp
class Solution
{
    public:
    //Function to find a continuous sub-array which adds up to a given number.
    vector<int> subarraySum(vector<int>arr, int n, long long s)
    {
        // Your code here
        int sum = 0,l=0,r=0;
        while(r<n){
            sum+= arr[r];
            while(sum>s && l<r){
                sum-=arr[l];
                l++;
            }
            if(sum==s){
                return {l+1,r+1};
            }
             r++;
        }
        return {-1};
    }
};

File: Arrays/Subarray_with_Given_Sum.py
class Solution:
    def subArraySum(self,arr, n, s): 
       #Write your code here
        sum ,l,r = 0,0,0
        
        while r<n:
          sum+=arr[r]
          
          while sum>s and l < r:
           sum-=arr[l]
           l+=1
          if sum==s:
            return (l+1,r+1)
          r+=1
          
        return (-1,)
           
